### PR TITLE
Ignore non .rs files for tidy libcoretest

### PR DESCRIPTION
**Pull request by varkor**
Previously, any file would be read, which is both unnecessary, and causes issues if irrelevant non-Unicode files were read (e.g. `.DS_STORE`).

### CODE REVIEW

File: src/tools/tidy/src/libcoretest.rs [hunk 1]
```diff
@@ -22,12 +22,15 @@ pub fn check(path: &Path, bad: &mut bool) {
         &libcore_path,
         &mut |subpath| t!(subpath.strip_prefix(&libcore_path)).starts_with("tests"),
         &mut |subpath| {
-            if t!(read_to_string(subpath)).contains("#[test]") {
-                tidy_error!(
-                    bad,
-                    "{} contains #[test]; libcore tests must be placed inside `src/libcore/tests/`",
-                    subpath.display()
-                );
+            if subpath.ends_with(".rs") {
```
Comment (alexcrichton) [inline src/tools/tidy/src/libcoretest.rs]: Have you verified that adding `#[test]` inside libcore still triggers this warning? `Path::ends_with` I don't think will work with `".rs"` here

Comment (varkor) [reply src/tools/tidy/src/libcoretest.rs]: Sorry, should have tested this properly. I've fixed it, and also improved the error message in the case of failure.

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

**Pull request merged**